tools/functional-tester/etcd-runner/main.go

```diff
@@ -212,6 +219,166 @@ func runRacer(eps []string, round int) {
 	doRounds(rcs, round)
 }
 
+func runWatcher(eps []string) {
+	var runningTime time.Duration = 30 // time for which operation should be performed
+	noOfPrefixes := 36                 // total number of prefixes which will be watched upon
+	watchPerPrefix := 10               // number of watchers per prefix
+	reqRate := 20                      // put request per second
+	keyPrePrefix := 100000             // max number of keyPrePrefixs for put operation
+
+	ctx := context.Background()
+	prefixKeysMap := make(map[string][]string) // map for holding prefixes and values
+	// pre generate prefixes equal to noOfPrefixes
+	prefixes := generateUniqueKeys(5, noOfPrefixes)
+	// pre generate keys
+	keys := generateRandomKeys(10, keyPrePrefix)
+
+	var (
+		revision int64
+		wg       sync.WaitGroup
+		pr       *clientv3.PutResponse
+		gr       *clientv3.GetResponse
+		err      error
+	)
+
+	for i := range prefixes {
+		prefix := prefixes[i]
+		// initialize prefix key map for each prefix
+		prefixKeysMap[prefix] = make([]string, 0)
```

Comment:
@xiang90 we cannot put initialization of all keys in a map and setting of all keys in the map in the same loop since the map needs to be initialized for all prefixes before we start putting any key to the map.
The reason being for example we have prefixes a, ab and abc, but the map does not have any entries, and we get prePrefix key as bcd and we get  prefix a to it to form the final key. so the final key becomes abcd, but it gets added only for prefix a, since they other keys are not initialized in map. But when we put the key abcd in put it will be retrieved by watchers with prefixes a, ab and abcd. Hence on comparison test will fail.
Hence all prefixes should be initialized in map before we start putting any key in a map.
What do you suggest?